connmgr/connmanager.go

```diff
@@ -429,6 +428,13 @@ out:
 	log.Trace("Connection handler done")
 }
 
+// ResetFailedAttempts sets the failed connection attempts counter
+// to zero. This method is expected to be called after every time
+// a peer had been successfully connected and marked as good.
+func (cm *ConnManager) ResetFailedAttempts() {
```

Comment:
this name exposes implementation details. Instead I would call it "ConnectionRequestComplete" or something like this